tests/CsvIteratorTest.php

```diff
@@ -0,0 +1,126 @@
+<?php
+
+namespace RonasIT\Support\Tests;
+use ErrorException;
+use ReflectionProperty;
+use RonasIT\Support\Iterators\CsvIterator;
+
+class CsvIteratorTest extends HelpersTestCase
+{
+    protected CsvIterator $csvIteratorClass;
+    protected ReflectionProperty $columnsProperty;
+
+    public function setUp(): void
+    {
+        parent::setUp();
+
+        $this->csvIteratorClass = new CsvIterator($this->getFixturePath('addresses.csv'));
+
+        $this->columnsProperty = new ReflectionProperty(CsvIterator::class, 'columns');
+        $this->columnsProperty->setAccessible(true);
+    }
+
+    public function testOpenFile()
+    {
+        $this->expectException(ErrorException::class);
+
+        $this->getMockBuilder(CsvIterator::class)
+            ->setConstructorArgs(['not_exists_file.csv'])
+            ->getMock();
+
+        $this->expectExceptionMessage('fopen(not_exists_file.csv): failed to open stream: No such file or directory');
+    }
+
+    public function testParseColumns()
+    {
+        $header = $this->getJsonFixture('header.json');
+
+        $this->csvIteratorClass->parseColumns($header);
```

Comment:
let's deprecate the `parseColumns` method and implement `setColumns`

Reply:
@DenTray Lets move it into separate task with other refactoring related to CsvIterator which I mentioned in our previous discussion